### ISSUE TITLE
Enable color-contrast accessibility test and fix all violations

### ISSUE THREAD

**Issue by carols10cents** [rust-lang/crates.io#1233]
Once #1227 is merged in, address [this comment](https://github.com/rust-lang/crates.io/pull/1227/commits/ed4333ae06fbfe9ad769901eaffa070b9265430e#r160559295):

> There are multiple elements that currently fail this test. I have disabled it since I think it needs to be addressed by someone more inclined towards design.

by changing tests/axe-config.js to remove:

```
'color-contrast': {
            enabled: false,
}
```

run `npm test` and make changes to fix all violations where we're using colors that don't contrast enough.

**Comment by sorin-davidoi** [comment]
[Colorable](https://colorable.jxnblk.com/) can be handy when testing new colour combinations.

**Comment by p-jackson** [comment]
I had this idea to use font size and weight to communicate information hierarchy rather than using grey (specifically `$main-color-light`)

![image](https://user-images.githubusercontent.com/1500769/51533858-efeaeb00-1ea8-11e9-984b-800137c72657.png)

Notice the changes to the version number in the heading and the metadata titles on the right.

Dealing with the hyperlink colour is trickier. It's very hard to pick greens with sufficient contrast. There's a reason why blue hyperlinks are ubiquitous ;)
